tree.go

```diff
@@ -22,33 +22,56 @@ type Tree interface {
 	Root() ([]byte, error)
 }
 
-var _ Tree = &DefaultTree{}
+// treeFns is a global map used for keeping track of registered tree constructors for JSON serialization
+// The keys of this map should be kebab cased. E.g. "default-tree"
+var treeFns = sync.Map{}
+
+// Must be called in the init function
+func RegisterTree(treeName string, treeConstructor TreeConstructorFn) error {
```

Comment:
Same here. Also, It would have been a must only in case we avoided synchronization. We use a sync map there now, so users can call registered whenever they like. We can only recommend it to be called in init, but not a must anymore.